beta/src/components/MDX/Sandpack/CustomPreset.tsx

```diff
@@ -54,59 +51,57 @@ export function CustomPreset({
-                onClick={() => {
-                  const nextIsExpanded = !isExpanded;
-                  flushSync(() => {
-                    setIsExpanded(nextIsExpanded);
-                  });
-                  if (!nextIsExpanded && containerRef.current !== null) {
-                    scrollIntoView(containerRef.current, {
-                      scrollMode: 'if-needed',
-                      block: 'nearest',
-                      inline: 'nearest',
-                    });
-                  }
-                }}>
-                <span className="flex p-2 focus:outline-none text-primary dark:text-primary-dark">
-                  <IconChevron
-                    className="inline mr-1.5 text-xl"
-                    displayDirection={isExpanded ? 'up' : 'down'}
-                  />
-                  {isExpanded ? 'Show less' : 'Show more'}
-                </span>
-              </button>
-            )}
-          </div>
 
-          {showDevTools && (
-            <SandpackReactDevTools onLoadModule={onDevToolsLoad} />
+        <SandpackLayout
+          className={cn(
+            showDevTools && devToolsLoaded && 'sp-layout-devtools',
+            !isExpandable && 'rounded-b-lg overflow-hidden',
+            isExpanded && 'sp-layout-expanded'
+          )}>
+          <SandpackCodeEditor
+            showLineNumbers
+            showInlineErrors
+            showTabs={false}
+            showRunButton={false}
+            extensions={lintExtensions}
+          />
+          <Preview
+            className="order-last xl:order-2"
+            isExpanded={isExpanded}
+            lintErrors={lintErrors}
+          />
+          {isExpandable && (
+            <button
+              translate="yes"
+              className="sandpack-expand flex text-base justify-between dark:border-card-dark bg-wash dark:bg-card-dark items-center z-10 p-1 w-full order-2 xl:order-last border-b-1 relative top-0"
+              onClick={() => {
+                const nextIsExpanded = !isExpanded;
+                flushSync(() => {
+                  setIsExpanded(nextIsExpanded);
+                });
+                if (!nextIsExpanded && containerRef.current !== null) {
+                  scrollIntoView(containerRef.current, {
+                    scrollMode: 'if-needed',
+                    block: 'nearest',
+                    inline: 'nearest',
+                  });
+                }
+              }}>
+              <span className="flex p-2 focus:outline-none text-primary dark:text-primary-dark">
+                <IconChevron
+                  className="inline mr-1.5 text-xl"
+                  displayDirection={isExpanded ? 'up' : 'down'}
+                />
+                {isExpanded ? 'Show less' : 'Show more'}
+              </span>
+            </button>
           )}
-        </SandpackThemeProvider>
+        </SandpackLayout>
+
+        {showDevTools && (
+          // @ts-ignore TODO(@danilowoz): support devtools
```

Comment:
TODO: reimplement `react-devtools-inline` support

Reply:
Shall we create an issue so that we don't miss this one?

Reply:
I have a PR waiting to be merged, once we merge this one.

https://github.com/codesandbox/sandpack/pull/537